### ISSUE TITLE
Refers needs to be a pointer

### ISSUE THREAD

**Issue by sudo-bmitch** [oci-playground/artifact-spec#3]
`omitempty` doesn't work on empty structs. This will result in a `refers: {}` output when the manifest is marshaled. This should be converted to a pointer.

https://github.com/oci-playground/artifact-spec/blob/eb1cd572d3def3a25a4f1488a242e3a527bfddaf/specs-go/v1/artifact.go#L29

**Comment by sajayantony** [comment]
@sudo-bmitch would you like to update this since it’s just a place holder for now?

**Comment by sudo-bmitch** [comment]
@sajayantony I'm good with pushing the PR. Just wanted to coordinate with you for any impact it has on oci-playground/distribution.

**Comment by sajayantony** [comment]
We can iterate. I think we also need to update https://github.com/oci-playground/image-spec/blob/1a83d467956c64aa8abfa02b4623e84d73e86dbd/specs-go/v1/manifest.go#L37

/cc @akashsinghal

**Comment by sudo-bmitch** [comment]
@sajayantony yup. I've got https://github.com/oci-playground/image-spec/issues/1 open over there.

**Comment by sudo-bmitch** [comment]
Switching to a pointer changed a lot of my regclient code, both creating pointers instead of structs, and adding nil checks. So when the go mod dependency is bumped, I'd expect you'll need to do similar.
